screen-wake-lock/wakelock-onrelease.https.html

```diff
@@ -20,4 +20,19 @@
     assert_false(ev.cancelable);
   });
 }, "Test onreleased event's basic properties");
+
+promise_test(async t => {
+  await test_driver.set_permission({ name: 'screen-wake-lock' }, 'granted', false);
+
+  const lock = await navigator.wakeLock.request("screen");
+
+  let onreleaseCalled = false;
+  lock.onrelease = t.step_func(() => {
+    onreleaseCalled = true;
+  });
+
+  return lock.release().then(() => {
+    assert_true(onreleaseCalled, "The 'release' event must be emitted before release() resolves");
+  });
```

Comment:
Would this make sense: 

```suggestion
  let releasedFired = false;
  lock.onrelease = () => {
     releasedFired = true;
  };
  
  // causes event to fire
  const releasePromise = lock.release();
  
  assert_true(releasedFired, "The 'release' event fires immediately after release() is called");
  await releasePromise;
```



If I'm reading https://github.com/w3c/screen-wake-lock/pull/299/files#diff-0eb547304658805aad788d320f10bf1f292797b5e6d745a3bf617584da017051R756 , then the event fires immediately.

Reply:
Yes, it would, thanks. I've updated the PR with the suggestion and some small changes.